packages/liveblocks-core/src/crdts/LiveList.ts

```diff
@@ -585,31 +585,44 @@ export class LiveList<TItem extends Lson> extends AbstractCrdt {
       throw new Error("Can't attach child if managed pool is not present");
     }
 
-    let result: ApplyResult;
+    const result = this._dispatchByIntent(op, source);
+    if (result.modified !== false) {
+      this.invalidate();
+    }
+
+    return result;
+  }
 
+  _dispatchByIntent(op: CreateChildOp, source: OpSource): ApplyResult {
     if (op.intent === "set") {
       if (source === OpSource.REMOTE) {
-        result = this._applySetRemote(op);
-      } else if (source === OpSource.ACK) {
-        result = this._applySetAck(op);
+        return this._applySetRemote(op);
+      } else if (
+        source === OpSource.PURE_ACK ||
+        source === OpSource.PATCHED_ACK
```

Comment:
I've changed all call sites that used `ACK` to use _either_ of these two, but I think what we'll want to do is to go over each use case, and decide which of the two cases we are intending to handle here. My assumption is that we're not interested in `PURE_ACK`, and only want to handle `PATCHED_ACK`, but let's talk about this in depth.